.github/workflows/simple.yml

```diff
@@ -44,3 +44,24 @@ jobs:
 
       - name: Test
         run: cabal test all --enable-tests --test-show-details=direct
+  wasi:
+    runs-on: ubuntu-latest
+    steps:
+    - name: setup-ghc-wasm32-wasi
+      run: |
+        cd $(mktemp -d)
+        curl -L https://gitlab.haskell.org/ghc/ghc-wasm-meta/-/archive/master/ghc-wasm-meta-master.tar.gz | tar xz --strip-components=1
```

Comment:
is this pinned bindist? I don't want "moving targets" in CI.

Reply:
Ah, just noticed your edit; I don't think it is easy ATM to install it via ghcup as you need to install a WASI-specific toolchain in addition to the bindist. Personally, I use ghc-wasm-meta via Nix, but there are also [community-provided Docker containers](https://discourse.haskell.org/t/dockerfiles-for-js-and-wasm-backends/5994).

Reply:
Aren't WASI toolchains available as `apt` packages? I wish I could get GHC from `apt` repository instead of `ghcup` too. I guess every community has their own way to install their stuff nowadays. I honestly don't like that, as one has to (get and) learn N install tools.

Reply:
E.g. wasi-sdk seems not to be packaged in Debian/Ubuntu (e.g. via [Repology](https://repology.org/project/wasi-sdk/versions)); and the GHC WASM backend needs a [wasi-sdk fork](https://gitlab.haskell.org/ghc/wasi-sdk) anyway.

Pinging @TerrorJack in case you have an estimate whether/when a ghcup- or apt-based installation of the WASM backend could become feasible.

Reply:
@phadej If you prefer, you can pin to a specific `ghc-wasm-meta` revision, since each revision pins to specific bindists, and that'll make the installed cross compiler binary-reproducible. @amesgen Would you do that in your next push? Thanks :)

We can't use vanilla `wasm32-wasi` toolchains from upstreams yet due to certain necessary changes in our fork. I understand the reluctance of using another installation method, but getting `ghcup` to work with cross toolchains would be a quite non-trivial amount of work that I haven't gotten enough time to implement yet 🙏

Reply:
Compared to `ghcup`, it'll be much easier to have an aptitude repo that distributes these tools. I'll look into the possibility this week, though no ETA can be quickly given at this point